file-handling/dirwatcher.js

```diff
@@ -0,0 +1,75 @@
+import { EventEmitter } from 'events';
+import fs from 'fs';
+import { promisify } from 'util';
+
+const readdirAsync = promisify(fs.readdir);
+
+//used by alternative version
+//const statAsync = promisify(fs.stat);
+
+export class DirWatcher extends EventEmitter {
+    constructor() {
+        super();
+        this.processedFiles = [];
+    }
+
+    watch(path, delay) {
+
+        //using setTimeout approach to make sure that dirwatcher is able to handle arbitrary number of CSV files without overlapping
+        const action = async () => {
+
+            const files = await readdirAsync(path);
+
+            await Promise.all(files.map(async (filename) => {
```

Comment:
async is not needed if there is no await inside a function

Reply:
required again after latest commit